src/common/Microsoft.DotNet.UpgradeAssistant.Abstractions/ProjectItemMatcher.cs

```diff
@@ -33,12 +33,19 @@ public ProjectItemMatcher(string match)
         }
 
         public bool Match(string input)
-            => _match switch
+        {
+            if (input is null)
+            {
+                throw new ArgumentNullException(nameof(input));
+            }
+
+            return _match switch
             {
                 Regex regex => regex.IsMatch(Path.GetFileName(input)),
-                string str => str.EndsWith(input, StringComparison.OrdinalIgnoreCase),
+                string str => input.EndsWith(str, StringComparison.OrdinalIgnoreCase),
```

Comment:
This is the one functional change to Upgrade Assistant in this PR. Based on my understanding of the item matcher, I think this check was backwards. I didn't write this, though, so if anyone knows why it was the way it was, let me know!

Reply:
I wrote this :) not sure, but this seems right. I'm surprised it didn't change anything else!

We should add some tests here

Reply:
Yea, I was surprised too! But all the tests still pass. I took a look and it looks like the other places where we call with a string, we were comparing the full string.